src/main/kotlin/app/hashers/CommitCrawler.kt

```diff
@@ -14,137 +14,168 @@ import io.reactivex.Observable
 import org.eclipse.jgit.api.Git
 import org.eclipse.jgit.diff.DiffEntry
 import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.EditList
 import org.eclipse.jgit.diff.RawText
 import org.eclipse.jgit.errors.MissingObjectException
 import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.Repository
 import org.eclipse.jgit.revwalk.RevCommit
 import org.eclipse.jgit.revwalk.RevWalk
 import org.eclipse.jgit.util.io.DisabledOutputStream
 
+/**
+* Iterates over the diffs between commits in the repo's history.
+*/
 object CommitCrawler {
-    fun getObservable(git: Git, repo: Repo,
-                      numCommits: Int = 0): Observable<Commit> {
-        var curNumCommits = 0
-        return Observable
-            .create<Commit> { subscriber ->
-                try {
-                    val revWalk = RevWalk(git.repository)
-                    val commitId = git.repository.resolve(RepoHelper.MASTER_BRANCH)
-                    revWalk.markStart(revWalk.parseCommit(commitId))
-                    for (revCommit in revWalk) {
-                        subscriber.onNext(Commit(revCommit))
+    fun getJGitObservable(git: Git, tail : RevCommit? = null) :
+        Observable<Pair<RevCommit, List<Pair<DiffEntry, EditList>>>> =
```

Comment:
Use instead:
```
data class JgitPair(val commit: RevCommit, val list: List<String>)
data class JgitDiff(val diffEntry: DiffEntry, val editList: EditList)
...
Observable<JgitPair> =
```
Requires no code change, performance is the same, more readable, shorter name

Reply:
oh, yeah, I was thinking about typedefs, but wasn't sure how to make that in kotlin, thanks :)